ecosystem/nft-metadata-crawler/nft-metadata-crawler-utils/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright © Aptos Foundation
+
+use anyhow::Context;
+use chrono::{NaiveDateTime, Utc};
+use google_cloud_auth::{
+    project::{create_token_source, Config},
+    token_source::TokenSource,
+};
+use serde::Deserialize;
+use std::{fs::File, io::Read};
+
+pub mod gcs;
+pub mod pubsub;
+
+// Struct to help with parsing of CSV
+#[derive(Clone, Debug)]
+pub struct NFTMetadataCrawlerEntry {
+    pub token_data_id: String,
+    pub token_uri: String,
+    pub last_transaction_version: i32,
+    pub last_transaction_timestamp: chrono::NaiveDateTime,
+    pub last_updated: chrono::NaiveDateTime,
+    pub force: bool,
+}
+
+impl NFTMetadataCrawlerEntry {
+    pub fn new(s: String) -> anyhow::Result<Self> {
+        let parts: Vec<&str> = s.split(',').collect();
+        Ok(Self {
+            token_data_id: parts[0].to_string(),
+            token_uri: parts[1].to_string(),
+            last_transaction_version: parts[2].to_string().parse()?,
+            last_transaction_timestamp: NaiveDateTime::parse_from_str(
+                parts[3],
+                "%Y-%m-%d %H:%M:%S %Z",
+            )
+            .unwrap_or(NaiveDateTime::parse_from_str(
+                parts[3],
+                "%Y-%m-%d %H:%M:%S%.f %Z",
+            )?),
+            last_updated: Utc::now().naive_utc(),
+            force: parts[4].parse::<bool>().unwrap_or(false),
+        })
+    }
```

Comment:
Why are we doing the split here instead of first splitting then adding the entry? It's an anti pattern to have the instantiation function do validation. You likely want to do validation somewhere else and have clearer logs.

Reply:
Also please add explicit logs so that we know what s is. Ideally you'd add some info about the task itself so that we can find it in the pub sub queue.